aten/src/ATen/native/cpu/TensorCompareKernel.cpp

```diff
@@ -143,6 +143,50 @@ static void max_kernel_impl(
   });
 }
 
+static void _min_max_val_kernel_impl(
+    Tensor& min_result,
+    Tensor& max_result,
+    const Tensor& self,
+    int64_t dim,
+    bool keepdim) {
+  auto wrap_dim = maybe_wrap_dim(dim, self.dim());
+  int64_t self_dim_size = ensure_nonempty_size(self, wrap_dim);
+
+  TORCH_CHECK(min_result.scalar_type() == self.scalar_type() && max_result.scalar_type() == self.scalar_type(),
+    "Expect min and max dtype ", self.scalar_type(),
+    " but got ", min_result.scalar_type(), " and ", max_result.scalar_type());
+
+  AT_DISPATCH_ALL_TYPES_AND(ScalarType::Bool, self.scalar_type(), "_min_max_val_cpu", [&] {
+    compare_base_kernel<scalar_t, scalar_t>(min_result, max_result, self, wrap_dim, keepdim, [&] (
+      scalar_t* min_result_data, scalar_t* max_result_data,
+      const scalar_t* self_data, auto self_dim_stride) {
+        using value_t = typename c10::scalar_value_type<scalar_t>::type;
+        scalar_t min_number = self_data[0];
+        scalar_t max_number = self_data[0];
+        for (int64_t i = 0; i < self_dim_size; ++i) {
+          scalar_t value = self_data[i * self_dim_stride];
+          // note: comparison is written this way to handle NaN correctly
+          if (!(value >= min_number)) {
+            min_number = value;
+            if (_isnan<scalar_t>(value)) {
+              max_number = value;
+              break;
+            }
+          } else if (!(value <= max_number)) {
+            max_number = value;
+            if (_isnan<scalar_t>(value)) {
```

Comment:
you can't ever get to this branch, `nan` is handled above

Reply:
thanks, fixed!